tests/lang_es_test.rs

```diff
@@ -0,0 +1,88 @@
+    };
+    driver.to_words()
+}
+#[test]
+fn test_lang_es() {
+    let prefs_basics =
+        ["negativo" /* , "veinte", "menos", "prepended", "appended", "bajo cero" */];
+    let prefs_for_ordinals = vec!["femenino" /* "f", "feminine", */, "plural"];
+    let prefs_for_decimal_char = vec!["coma"];
+
+    let driver = |output: Outputs, num: BigFloat| {
+        to_words(
+            num,
+            output,
+            prefs_basics
+                .iter()
+                .chain(&prefs_for_decimal_char)
+                .copied()
+                .collect::<Vec<_>>()
+                .as_slice(),
+        )
+    };
+    let word = driver(Outputs::Cardinal, BigFloat::from(-821_442_524.69)).unwrap();
+    assert_eq!(
+        word,
+        "ochocientos veintiún millones cuatrocientos cuarenta y dos mil quinientos veinticuatro \
+         coma seis nueve negativo"
+    );
+    let word = driver(Outputs::Ordinal, BigFloat::from(-484));
+    assert!(word.is_err()); // You can't get the ordinal of a negative number
+
+    let driver =
+        |output: Outputs, num: BigFloat| to_words(num, output, prefs_for_ordinals.as_slice());
+
+    // let driver = to_language(Lang::Spanish, prefs_for_ordinals.clone());
+    // let word = ;
+    assert_eq!(driver(Outputs::Ordinal, BigFloat::from(14)).unwrap(), "decimocuartas");
+    assert_eq!(driver(Outputs::Ordinal, BigFloat::from(1)).unwrap(), "primeras");
+    assert_eq!(driver(Outputs::Ordinal, BigFloat::from(2)).unwrap(), "segundas");
+
+    let driver = |output: Outputs, num: BigFloat| to_words(num, output, &[]);
+    let word = driver(Outputs::Ordinal, BigFloat::from(141_100_211_021u64)).unwrap();
+    assert_eq!(
+        word,
+        "centésimo cuadragésimo primero billonésimo centésimo millonésimo ducentésimo undécimo \
+         milésimo vigésimo primero"
+    );
+    assert_eq!(driver(Outputs::Ordinal, BigFloat::from(14)).unwrap(), "decimocuarto");
+    assert_eq!(driver(Outputs::Ordinal, BigFloat::from(1)).unwrap(), "primero");
+    assert_eq!(driver(Outputs::Ordinal, BigFloat::from(2)).unwrap(), "segundo");
+    assert_eq!(driver(Outputs::Ordinal, BigFloat::from(3)).unwrap(), "tercero");
+
+    let word = to_words(BigFloat::from(14), Outputs::OrdinalNum, &["f"]);
+    assert_eq!(word.unwrap(), "14ª");
+    let word = to_words(BigFloat::from(14), Outputs::OrdinalNum, &[]);
+    assert_eq!(word.unwrap(), "14º");
+
+    let word = to_words(BigFloat::from(2021), Outputs::Year, &[]);
+    assert_eq!(word.unwrap(), "dos mil veintiuno");
+    let word = to_words(BigFloat::from(-2021), Outputs::Year, &[]);
+    assert_eq!(word.unwrap(), "dos mil veintiuno a. C.");
+
+    let word = to_words(BigFloat::from(21_001.21), Outputs::Currency, &[]);
+    assert_eq!(word.unwrap(), "veintiún mil un US dollars con veintiún centavos");
+
+    let word = to_words(BigFloat::from(21.01), Outputs::Currency, &[]);
+    assert_eq!(word.unwrap(), "veintiún US dollars con un centavo");
+}
```

Comment:
btw, compared the results from es.rs to python num2words and see differences

I don't mind much the differences in megas as it depends a lot on the source. however `mil trillones` makes me wonder, is this what should actually be the number?

```
10000000000000
diez billones
diez trillones

100000000000000
cien billones
cien trillones

1000000000000000000
un trillón
un quintillón

1000000000000000000000 <=== here
mil trillones
un sextillón

10000000000000000000000000
diez cuatrillones
diez septillones
```

Reply:
What's different?. Is Mil Trillones from Python?

I followed the standard of Each thousands changes the Millard because It was a bit confusing that a Billion*1000 is called the equivalent in spanish of "thousand billions" instead of "trillion"

Reply:
https://es.wikipedia.org/wiki/Anexo:Nombres_de_los_n%C3%BAmeros_en_espa%C3%B1ol
So according to this source, it starts to jump by 1_000_000 instead of 1_000 after the trillion

As summary, English's [Sextillion](https://en.wikipedia.org/wiki/English_numerals#Very_large_numbers) is 10^21
Spanish's Sextillion (according to [this particular page in wiki](https://es.wikipedia.org/wiki/Sextill%C3%B3n)) is 10^36

I guess it would still best to discuss with someone more familiarized with spanish, because if this really is the spanish standard, I'm pretty sure almost none will get spanish's quadrillon right (Because it actually is "A Million Trillion")